idp-service/src/main/java/io/mosip/idp/services/KafkaHelperService.java

```diff
@@ -0,0 +1,72 @@
+
+import io.mosip.idp.core.exception.IdPException;
+import io.mosip.idp.core.util.ErrorConstants;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.config.KafkaListenerEndpointRegistry;
+import org.springframework.kafka.config.MethodKafkaListenerEndpoint;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.listener.MessageListener;
+import org.springframework.kafka.listener.MessageListenerContainer;
+import org.springframework.kafka.listener.adapter.RecordFilterStrategy;
+import org.springframework.messaging.handler.annotation.support.DefaultMessageHandlerMethodFactory;
+import org.springframework.stereotype.Component;
+
+import javax.validation.constraints.NotNull;
+
+@Slf4j
+@Component
+public class KafkaHelperService {
+
+    @Autowired
+    private KafkaTemplate<String,String> kafkaTemplate;
+
+    @Autowired
+    private KafkaListenerEndpointRegistry kafkaListenerEndpointRegistry;
+
+    @Autowired
+    private KafkaListenerContainerFactory kafkaListenerContainerFactory;
+
+    public void publish(@NotNull String topic, @NotNull String message) {
+        kafkaTemplate.send(topic, message);
+        log.info("Published message to topic : {}", topic);
+    }
+
+    public void subscribe(@NotNull String topic, @NotNull String id,
+                          @NotNull MessageListener messageListener, RecordFilterStrategy recordFilterStrategy)
+            throws IdPException {
+        MethodKafkaListenerEndpoint<String, String> kafkaListenerEndpoint = new MethodKafkaListenerEndpoint<>();
+        kafkaListenerEndpoint.setId(id);
+        kafkaListenerEndpoint.setTopics(topic);
+        kafkaListenerEndpoint.setAutoStartup(true);
+        kafkaListenerEndpoint.setMessageHandlerMethodFactory(new DefaultMessageHandlerMethodFactory());
+        kafkaListenerEndpoint.setBean(messageListener);
+        kafkaListenerEndpoint.setRecordFilterStrategy(recordFilterStrategy);
+
+        try {
+            kafkaListenerEndpoint.setMethod(messageListener.getClass().getMethod("onMessage", ConsumerRecord.class));
+        } catch (NoSuchMethodException e) {
+             log.error("Failed to find method 'onMessage' in the provided MessageListener instance!", e);
+             throw new IdPException(ErrorConstants.SUBSCRIBE_FAILED);
+        }
+        kafkaListenerEndpointRegistry.registerListenerContainer(kafkaListenerEndpoint,
+                kafkaListenerContainerFactory, true);
+    }
+
+    public void unsubscribe(@NotNull String id) {
+        MessageListenerContainer listenerContainer = kafkaListenerEndpointRegistry.getListenerContainer(id);
+        if(listenerContainer != null) {
+            listenerContainer.stop();
+            log.info("Stopped the kafka listener with id : {}", id);
+        }
```

Comment:
Do we still need the subscribe and unsubscribe methods here ?

Reply:
Not used anywhere, removed.